mailpoet/assets/js/src/mailpoet-email-editor-integration/middleware/strip-post-status-on-save.ts

```diff
@@ -0,0 +1,148 @@
+import { store as coreDataStore } from '@wordpress/core-data';
+import { store as editorStore } from '@wordpress/editor';
+import { use } from '@wordpress/data';
+import { storeName as emailEditorStore } from '@woocommerce/email-editor';
+
+// Define types for better type safety
+type Registry = {
+  select: (store: string) => Record<string, (...args: unknown[]) => unknown>;
+  dispatch: (store: string) => Record<string, (...args: unknown[]) => unknown>;
+};
+
+type ActionFunction = (...args: unknown[]) => Promise<unknown>;
+type ActionsRecord = Record<string, ActionFunction>;
+type OriginalActionsStore = Record<string, Record<string, ActionFunction>>;
+
+// Keep track of original actions per store
+const originalActions: OriginalActionsStore = {};
+// Keep information about initialization
+let isInitialized = false;
+
+// Which store and actions to wrap
+const INTERCEPTED_ACTIONS: Record<string, string[]> = {
+  core: ['saveEntityRecord'],
+};
+
+/**
+ * Handles logic of processing and dispatching the stripped post status.
+ *
+ * @param args           The arguments passed to the original action.
+ * @param registry       The data registry for use during processing.
+ * @param originalAction The original action to call if the conditions are not met.
+ * @return The result of the original action or a custom process response.
+ */
+async function processAndDispatchStrippedStatus(
+  args: unknown[],
+  registry: Registry,
+  originalAction: ActionFunction,
+): Promise<unknown> {
+  try {
+    const [kind, name, recordOrId, options] = args;
+
+    // Validate kind and name
+    if (typeof kind !== 'string' || typeof name !== 'string') {
+      return await originalAction(...args);
+    }
+
+    const postType = registry
+      .select(emailEditorStore)
+      .getEmailPostType() as string;
+
+    // Proceed only for correct kind/name and when stripping is enabled
+    if (kind !== 'postType' || name !== postType) {
+      return await originalAction(...args);
+    }
+
+    // Ensure recordOrId is object with numeric id
+    if (
+      typeof recordOrId !== 'object' ||
+      recordOrId === null ||
+      typeof (recordOrId as Record<string, unknown>).id !== 'number'
```

Comment:
```suggestion
      typeof (recordOrId as Record<string, unknown>)?.id !== 'number'
```

Also, if we are specifically checking for a number, we can typecast it to a numeric type for the check.
e.g.
```JavaScript
      typeof +recordOrId?.id !== 'number'
```